core/src/main/java/org/apache/calcite/rel/core/RelFactories.java

```diff
@@ -376,40 +371,12 @@ RelNode createCorrelate(RelNode left, RelNode right,
   private static class CorrelateFactoryImpl implements CorrelateFactory {
     public RelNode createCorrelate(RelNode left, RelNode right,
         CorrelationId correlationId, ImmutableBitSet requiredColumns,
-        SemiJoinType joinType) {
+        JoinRelType joinType) {
       return LogicalCorrelate.create(left, right, correlationId,
           requiredColumns, joinType);
     }
   }
 
-  /**
-   * Can create a semi-join of the appropriate type for a rule's calling
-   * convention.
-   */
-  public interface SemiJoinFactory {
```

Comment:
The SemiJoin class was not removed in this PR but it was deprecated. We should be consistent, and don't remove the factory but deprecate it first!

Reply:
Initially i thought they are internal APIs that only used for `RelBuilder`, but I agree keeping them will make better compatibility, add them back.